src/JsonSerializationVisitor.php

```diff
@@ -147,7 +147,7 @@ public function visitProperty(PropertyMetadata $metadata, $v): void
             return;
         }
 
-        if (true === $metadata->skipWhenEmpty && ($v instanceof \ArrayObject || \is_array($v)) && 0 === count($v)) {
+        if (true === $metadata->skipWhenEmpty && empty($v)) {
```

Comment:
It is not going to work with `ArrayObject`  as `empty` will always return `false` for any object. Try: `var_dump(empty(new \ArrayObject()));`

Reply:
Thank you @scyzoryck for checking it out! You are right, empty is not suitable on it's own,

What would you think about something like this:

`if (
    true === $metadata->skipWhenEmpty &&
    ((!is_object($v) && empty($v)) ||
        ($v instanceof \ArrayObject && 0 === count($v)))
) {`